tf_files/aws/modules/upload-data-bucket/s3.tf

```diff
@@ -0,0 +1,107 @@
+
+## The actual data bucket
+
+resource "aws_s3_bucket" "data_bucket" {
+  bucket = "${var.vpc_name}-data-bucket"
+  acl    = "private"
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  logging {
+    target_bucket = "${aws_s3_bucket.log_bucket.id}"
+    target_prefix = "log/${var.vpc_name}-data-bucket"
+  }
+
+  tags {
+    Name        = "${var.vpc_name}-data-bucket"
+    Environment = "${var.environment}"
+    Purpose     = "data bucket"
+  }
+}
+
+
+
+## Log bucket, where access to the avobe bucket will be logged 
+
+
+resource "aws_s3_bucket" "log_bucket" {
+  bucket = "${var.vpc_name}-data-bucket-logs"
+  acl    = "bucket-owner-full-control" #log-delivery-write
+  acl    = "log-delivery-write"
+
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  lifecycle_rule {
+    id      = "log"
+    enabled = true
+
+    prefix = "/"
+
+    tags {
+      "rule"      = "log"
+      "autoclean" = "true"
+    }
+
+    expiration {
+      days = 120
```

Comment:
we expire the logs here, because we know they'll also be logged in cloud-trail?

Reply:
Basically yes, they are also being sent over to CSOC, I don't think we necessarily need them redundant on both sides (commons account and CSOC), but this is and easy fix if we need them both up to date